server/sonar-server/src/main/java/org/sonar/server/telemetry/TelemetryDaemon.java

```diff
@@ -0,0 +1,187 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.telemetry;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import org.picocontainer.Startable;
+import org.sonar.api.config.Configuration;
+import org.sonar.api.platform.Server;
+import org.sonar.api.server.ServerSide;
+import org.sonar.api.utils.System2;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.api.utils.text.JsonWriter;
+import org.sonar.core.platform.PluginRepository;
+import org.sonar.server.es.SearchOptions;
+import org.sonar.server.measure.index.ProjectMeasuresIndex;
+import org.sonar.server.measure.index.ProjectMeasuresStatistics;
+import org.sonar.server.property.InternalProperties;
+import org.sonar.server.user.index.UserIndex;
+import org.sonar.server.user.index.UserQuery;
+
+import static org.sonar.api.measures.CoreMetrics.LINES_KEY;
+import static org.sonar.api.measures.CoreMetrics.NCLOC_KEY;
+import static org.sonar.api.utils.DateUtils.formatDate;
+import static org.sonar.api.utils.DateUtils.parseDate;
+import static org.sonar.core.config.TelemetryProperties.PROP_ENABLE;
+import static org.sonar.core.config.TelemetryProperties.PROP_FREQUENCY;
+import static org.sonar.core.config.TelemetryProperties.PROP_URL;
+
+@ServerSide
+public class TelemetryDaemon implements Startable {
+  private static final String THREAD_NAME_PREFIX = "sq-telemetry-service-";
+  private static final int SEVEN_DAYS = 7 * 24 * 60 * 60 * 1_000;
+  static final String I_PROP_LAST_PING = "telemetry.lastPing";
+  static final String I_PROP_OPT_OUT = "telemetry.optOut";
+  private static final Logger LOG = Loggers.get(TelemetryDaemon.class);
+
+  private final TelemetryClient telemetryClient;
+  private final Configuration config;
+  private final InternalProperties internalProperties;
+  private final Server server;
+  private final PluginRepository pluginRepository;
+  private final System2 system2;
+  private final UserIndex userIndex;
+  private final ProjectMeasuresIndex projectMeasuresIndex;
+
+  private ScheduledExecutorService executorService;
+
+  public TelemetryDaemon(TelemetryClient telemetryClient, Configuration config, InternalProperties internalProperties, Server server, PluginRepository pluginRepository,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)